File: test/aidbox_sdk/snapshots/typescripthl7-fhir-r4-core/SearchParameter.ts
import { UsageContext } from "./UsageContext";
import { ContactDetail } from "./ContactDetail";
import { CodeableConcept } from "./CodeableConcept";
import { DomainResource } from "./DomainResource";
import { BackboneElement } from "./BackboneElement";

export type SearchParameterComponent = BackboneElement & {
    definition: string;
    expression: string;
};

export type SearchParameter = DomainResource & {
    description: string;
    date?: string;
    expression?: string;
    modifier?: string[];
    publisher?: string;
    multipleAnd?: boolean;
    jurisdiction?: CodeableConcept[];
    derivedFrom?: string;
    purpose?: string;
    multipleOr?: boolean;
    name: string;
    useContext?: UsageContext[];
    xpath?: string;
    xpathUsage?: string;
    type: string;
    experimental?: boolean;
    component?: SearchParameterComponent[];
    status: string;
    chain?: string[];
    url: string;
    code: string;
    comparator?: string[];
    target?: string[];
    base: string[];
    version?: string;
    contact?: ContactDetail[];
};

File: test/aidbox_sdk/snapshots/python/hl7_fhir_r4_core/SearchParameter.py
from typing import Optional, List
from dataclasses import dataclass, field
from base import UsageContext
from base import ContactDetail
from base import CodeableConcept
from base import DomainResource
from base import BackboneElement

@dataclass(kw_only=True)
class SearchParameterComponent(BackboneElement):
    definition: str
    expression: str

@dataclass(kw_only=True)
class SearchParameter(DomainResource):
    description: str
    date: Optional[str] = None
    expression: Optional[str] = None
    modifier: Optional[List[str]] = field(default_factory=list)
    publisher: Optional[str] = None
    multiple_and: Optional[bool] = None
    jurisdiction: Optional[List[CodeableConcept]] = field(default_factory=list)
    derived_from: Optional[str] = None
    purpose: Optional[str] = None
    multiple_or: Optional[bool] = None
    name: str
    use_context: Optional[List[UsageContext]] = field(default_factory=list)
    xpath: Optional[str] = None
    xpath_usage: Optional[str] = None
    type: str
    experimental: Optional[bool] = None
    component: Optional[List[SearchParameterComponent]] = field(default_factory=list)
    status: str
    chain: Optional[List[str]] = field(default_factory=list)
    url: str
    code: str
    comparator: Optional[List[str]] = field(default_factory=list)
    target: Optional[List[str]] = field(default_factory=list)
    base: list[str] = field(default_factory=list)
    version: Optional[str] = None
    contact: Optional[List[ContactDetail]] = field(default_factory=list)